src/vs/code/electron-main/menus.ts

```diff
@@ -196,6 +200,15 @@ export class VSCodeMenu {
 			updateMenu = true;
 		}
 
+		let newAutoHideMenuBar = config && config.window && config.window.autoHideMenuBar;
```

Comment:
I think this should move out of `menu.ts` into `window.ts` because it rather belongs there. You can have the same listener installed in `window.ts` and do the updating. Also note that I think it should only update if the menu is actually hidden.